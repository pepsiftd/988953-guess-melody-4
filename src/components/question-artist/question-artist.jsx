import React from 'react';
import PropTypes from 'prop-types';

const AUDIO_PLAYER_ID = 0;

const QuestionArtist = ({question, onAnswer, renderPlayer}) => {
  const {answers, song} = question;

  return (
    <section className="game__screen">
      <h2 className="game__title">Кто исполняет эту песню?</h2>
      <div className="game__track">
        <div className="track">
          {renderPlayer(song.src, AUDIO_PLAYER_ID)}
        </div>
      </div>

      <form className="game__artist">
        {answers.map((answer, index) => {
          return (
            <div key={answer.artist} className="artist">
              <input
                className="artist__input visually-hidden"
                type="radio"
                name="answer"
                value={`artist-${index + 1}`}
                id={`answer-${index + 1}`}
                onChange={(evt) => {
                  evt.preventDefault();
                  onAnswer(question, answer);
                }}
              />
              <label className="artist__name" htmlFor={`answer-${index + 1}`}>
                <img className="artist__picture" src={answer.picture} alt={answer.artist}/>
                {answer.artist}
              </label>
            </div>
          );
        })}
      </form>
    </section>
  );
};

QuestionArtist.propTypes = {
  question: PropTypes.shape({
    song: PropTypes.shape({
      artist: PropTypes.string.isRequired,
      src: PropTypes.string.isRequired
    }).isRequired,
    answers: PropTypes.arrayOf(PropTypes.shape({
      picture: PropTypes.string.isRequired,
      artist: PropTypes.string.isRequired,
    })).isRequired
  }).isRequired,
  onAnswer: PropTypes.func.isRequired,
  renderPlayer: PropTypes.func.isRequired,
};

export {QuestionArtist};
