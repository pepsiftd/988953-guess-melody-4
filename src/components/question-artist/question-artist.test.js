import React from 'react';
import renderer from 'react-test-renderer';
import {QuestionArtist} from './question-artist';

const question = {
  type: `artist`,
  song: {
    artist: `Банд'эрос`,
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  },
  answers: [{
    picture: `http://placehold.it/134x134`,
    artist: `Space`,
  }, {
    picture: `http://placehold.it/134x134`,
    artist: `Бахыт компот`,
  }, {
    picture: `http://placehold.it/134x134`,
    artist: `Stereophonics`,
  }],
};

it(`QuestionArtist should render correctly`, () => {
  const tree = renderer.create(
      <QuestionArtist
        question={question}
        onAnswer={() => {}}
      />,
      {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
