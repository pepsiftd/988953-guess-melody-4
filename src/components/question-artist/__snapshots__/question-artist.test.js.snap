// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionArtist should render correctly 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    >
      <button
        className="track__button track__button--play"
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
    </div>
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="Space"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        Space
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Бахыт компот"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        Бахыт компот
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-3"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-3"
      />
      <label
        className="artist__name"
        htmlFor="answer-3"
      >
        <img
          alt="Stereophonics"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        Stereophonics
      </label>
    </div>
  </form>
</section>
`;
